internal/datastore/common/validation.go

```diff
@@ -0,0 +1,28 @@
+package common
+
+import (
+	"fmt"
+
+	v1 "github.com/authzed/authzed-go/proto/authzed/api/v1"
+
+	"github.com/authzed/spicedb/pkg/tuple"
+)
+
+// ValidateUpdatesToWrite performs basic validation on relationship updates going into datastores.
+func ValidateUpdatesToWrite(updates []*v1.RelationshipUpdate) error {
+	for _, update := range updates {
+		err := update.HandwrittenValidate()
+		if err != nil {
+			return err
+		}
+
+		if update.Relationship.Subject.Object.ObjectId == tuple.PublicWildcard && update.Relationship.Subject.OptionalRelation != "" {
+			return fmt.Errorf(
+				"Attempt to write a wildcard relationship (`%s`) with a non-empty relation. Please report this bug",
```

Comment:
We could put it into a proxy, but if that doesn't get installed (for whatever reason), then bad data could get into the system. I wanted it to be done via each handler to ensure its done right at the point before we write the relationships. I also did not have it panic because, well, I don't want it to crash the entire binary

Reply:
It's more likely that someone forgets they need to call this in their implementation of a data store than we miss adding it to the gRPC server. We already have absolutely required gRPC middleware -- either we trust it or we shouldn't bother having it.

Reply:
True, but I always tend to err on the side of extra checks where possible. Ideally, this would be a debug and testing-only check, but Go makes that less than fun to do

Reply:
We already have a validating datastore proxy for debug and testing checks. Put it there?